packages/expo-auth-session/src/AuthRequest.ts

```diff
@@ -263,7 +263,7 @@ export class AuthRequest implements Omit<AuthRequestConfig, 'state'> {
   }
 
   private async ensureCodeIsSetupAsync(): Promise<void> {
-    if (this.codeVerifier) {
+    if (this.codeChallenge) {
```

Comment:
Add a detailed comment here explaining the reasoning. In theory there shouldn't be anything crucial breaking, but it's hard to know for sure given the public scope of `codeVerifier`.

Reply:
Agreed. I'll try and get the local environment all set up and work through the flows in more detail. I'm new to this code though and oauth in general, so any feedback is welcome.

Essentially what this code currently seems to do is either generate both a verifier and a challenge, or allow the developer to pass in both their own verifier and challenge. 

Technically the developer could only pass in a verifier to stop the auto generation, however that would then not work as there would be a missing challenge on the code request.

I think changing this check to codeChallenge makes more sense as then the options are:

1) Pass in neither the verifier or the challenge and they are both generated for you. This means the PKCE flow is all done within the mobile client.

2) Pass in both a verifier and a challenge if the developer already has a pair generated locally, and still the PKCE flow is done with the mobile client.

3) Pass in only a challenge as the developer has received a challenge for a server endpoint, but the verifier is only known to the server, which will do the code for token exchange later server side.

Hopefully this makes sense. The other reason for this change is i lost a few hours passing in only a code challenge and kept getting 'Invalid code verifier' back from google as the library was auto generating a new pair. I can probably add some additional errors or warnings around this too.

Reply:
Another gap i've spotted is that AuthRequest doesn't even attempt to use the codeChallenge on AuthRequestConfig. I have now fixed this in the PR too and verified (with the google provider) that the flow now works if you use your own codeChallenge.